lib/membrane_wav/serializer.ex

```diff
@@ -72,16 +73,17 @@ defmodule Membrane.WAV.Serializer do
         :output,
         buffers_count,
         :buffers,
-        _context,
-        %{header_created: true, frames_per_buffer: frames, caps: caps} = state
+        context,
+        %{header_created: true, frames_per_buffer: frames} = state
       ) do
+    caps = context.pads.output.caps
     size = buffers_count * Caps.frames_to_bytes(frames, caps)
 
     {{:ok, demand: {:input, size}}, state}
   end
 
   @impl true
-  def handle_process(:input, buffer, _context, state) do
+  def handle_process(:input, buffer, _context, %{header_created: true} = state) do
```

Comment:
I'd still add an explicit case raising with human readable error - now it might not be obvious why the element is failing from "Function clause error"